apps/condo/domains/common/components/TrackingContext.tsx

```diff
@@ -0,0 +1,210 @@
+
+const TRACKING_INITIAL_VALUE = {
+    // Here you should create app related tracker instances
+    trackerInstances: { amplitude: new AmplitudeInstance() },
+}
+
+export type TrackingEventPropertiesType = {
+    page?: {
+        path: string
+        state?: string
+    }
+}
+
+type TrackingCommonEventProperties = Record<string, unknown>
+
+export interface ITrackingComponent {
+    eventName?: string
+    eventProperties?: TrackingCommonEventProperties
+}
+
+interface ITrackingContext {
+    trackerInstances: Record<string, TrackerInstance>
+    eventProperties?: TrackingEventPropertiesType
+    userProperties?: TrackingCommonEventProperties
+}
+
+interface ILogEventTo extends ITrackerLogEventType {
+    destination: Array<string>
+}
+
+const TrackingContext = createContext<ITrackingContext>(TRACKING_INITIAL_VALUE)
+
+const useTrackingContext = (): ITrackingContext => useContext<ITrackingContext>(TrackingContext)
+
+export enum TrackingEventType {
+    Visit = 'Visit',
+    Click = 'Click',
+    Input = 'Input',
+    Select = 'Select',
+    Checkbox = 'Checkbox',
+    Radio = 'Radio',
+    FollowExternalLink = 'FollowExternalLink',
+    ImportComplete = 'ImportComplete',
+}
+
+interface IUseTracking {
+    (): {
+        eventProperties: ITrackingContext['eventProperties']
+        userProperties: Pick<ITrackingContext, 'userProperties'>
+        logEvent: (logEventProps: ITrackerLogEventType) => void
+        logEventTo: (logEventToProps: ILogEventTo) => void
+        instrument: (eventName: string, eventProperties?: TrackingCommonEventProperties, func?: any) => any
+        getEventName: (eventType: TrackingEventType) => string
+    }
+}
+
+const useTracking: IUseTracking = () => {
+    const { route } = useRouter()
+    const { trackerInstances, eventProperties, userProperties } = useTrackingContext()
+
+    const logEvent = ({ eventName, eventProperties: localEventProperties = {}, denyDuplicates }: ITrackerLogEventType) => {
+        const resultEventProperties = {
+            ...eventProperties,
+            ...localEventProperties,
+        }
+        Object.values(trackerInstances).map((trackerInstance) => trackerInstance.logEvent({
+            eventName,
+            eventProperties: resultEventProperties,
+            denyDuplicates,
+            userProperties,
+        }))
+    }
+
+    const logEventTo = ({ eventName, eventProperties: localEventProperties = {}, destination, denyDuplicates }: ILogEventTo) => {
+        const resultEventProperties = {
+            ...eventProperties,
+            ...localEventProperties,
+        }
+        Object.values(pick(trackerInstances, destination)).map(trackerInstance => trackerInstance.logEvent({
+            eventName,
+            eventProperties: resultEventProperties,
+            denyDuplicates,
+        }))
+    }
+
+    const instrument = <T extends (...args: any[]) => any>(eventName: string, eventProperties?: TrackingEventPropertiesType, func?: T): T => {
```

Comment:
noun is a bad name for a function, better use `getXXXXXXX`

Reply:
Make name more clear to understanding